scripts/linters/html_linter.py

```diff
@@ -209,6 +217,46 @@ def handle_data(self, data):
             elif data_line.startswith(ending_block):
                 self.indentation_level -= 1
 
+    def _check_space_between_attributes_and_values(
+            self, tag, attr, value, rendered_text,
+            value_in_quotes, attr_pos_mapping):
+        """Checks if there are any spaces between attributes and their value.
+
+        Args:
+            tag: str. The tag name of the HTML line.
+            attr: str. The attribute name in the tag.
+            value: str. The value of the attribute.
+            rendered_text: str. The rendered text of the tag.
+            value_in_quotes: bool. Whether the given attribute value
+                is in double quotes.
+            attr_pos_mapping: dict. Mapping between attribute and their
+                starting positions in the tag.
+        """
+        line_number, _ = self.getpos()
+        if attr not in attr_pos_mapping:
+            attr_positions = []
+            # Finds the positions of the attribute in the tag.
+            for match in re.finditer(re.escape(attr), rendered_text.lower()):
+                start, end = match.start(), match.end()
+                # Appends the position only if it is an attribute.
+                # It will not append the position if it is a substring.
```

Comment:
We started off with making it simpler and ended up again with complicated checks with regex.

I'm fine with this lint-check, but wanted to check whether we can make it simpler with any other idea?
Maybe find index(i) of "=" in the start tag and value at i-1 or i+1 should not be space " "?

(I'm fine with approving this PR, just wanted to check whether we can simplify this!) :)

Reply:
@DubeySandeep I know right :P.
Well I did think of the idea you mentioned but the edge case for that are:

- `<button ... ng-click="$ctrl.isVisible = !$ctrl.isVisible">`
- `<button ... ng-click="$ctrl.explanationIsShown = true;">`

The reason why I used re.finditer was to find all the occurrences of that attribute in that tag.
We usually start with the simple solution but end up making it complicated when we fix all the edge cases :P

Actually, if we ignore the case where an attribute is defined twice. We can roll back to the previous version where it looked better. When a duplicate attribute comes, there arises a need to find all the positions of that attribute.

So in the case of `<div class="test" class = '"test">`
It will only throw an existing linter error of duplicate attribute

Reply:
do let me know what you think about it